crates/oxc_linter/src/config/config_store.rs

```diff
@@ -85,6 +85,13 @@ impl Config {
         let mut env = self.base.config.env.clone();
         let mut globals = self.base.config.globals.clone();
         let mut plugins = self.base.config.plugins;
+
+        for override_config in overrides_to_apply.clone() {
+            if let Some(override_plugins) = override_config.plugins {
+                plugins |= override_plugins;
+            }
+        }
+
```

Comment:
we need to loop over twice, once here to get the merged set of enabled plugins, so that we don't filter out rules before the plugin is added below